web/api/v1/api.go

```diff
@@ -1908,9 +1917,19 @@ func parseLimitParam(limitStr string) (limit int, err error) {
 	if err != nil {
 		return limit, err
 	}
-	if limit <= 0 {
-		return limit, errors.New("limit must be positive")
+	if limit < 0 {
+		return limit, errors.New("limit must be non-negative")
 	}
 
 	return limit, nil
 }
+
+// toStorageLimit increases the API limit, as returned by parseLimitParam, by 1.
+// This allows for emitting warnings when the results are truncated.
+func toStorageLimit(limit int) int {
```

Comment:
This feels a bit weird to me.
After some thought, I realized it allows the feature to work on storages which implement the limit and those which do not.  Is that why you did it?
Please write down why this behaviour exists.

Also I don't like the name; it's got nothing to do with "storage".  Maybe "toHintLimit" ?

Reply:
The warning `results truncated due to limit` is emitted in `api.go` only when the number of results returned from `storage.Querier` is greater than the limit.
To ensure that this logic works correctly, limit + 1 should be passed to the `storage.Querier`.

The intend with storage limit is to mean the limit that is passed to `storage.Querier`. I don't mind renaming it to `toHintLimit`.